feat: 재생 버튼 클릭 시 라디오 재생 로직 구현

Eun0713 opened the pull request:
### ✨ 이슈 번호 

- #44 

### 📌 설명 

- 오디오 스트리밍 재생 함수를 구현하고, MiniPlayer와 ChannelPlayer에 적용한 Pull Request입니다.

### 📃 작업 사항 

- [X] hls.js 라이브러리 설치
- [X] 스트리밍 재생/일시정지 유틸 함수(controlStreamPlayback) 구현
- [X] MiniPlayer에 스트리밍 재생 기능 연동
- [X] ChannelPlayer에 동일 기능 적용

### 💡 참고 사항 
**🔊 스트리밍 재생 시점에 콘솔에 로그가 출력**

![스트리밍 재생 로그 캡처](https://github.com/user-attachments/assets/6171270a-6b2f-40bf-b9ba-4a7ef180812d)

### 💭 리뷰 사항 반영 

### ✅ Pull Request 체크 사항

- [X] 가장 최신 브랜치를 pull 하였습니다.
- [X] base 브랜치명을 확인하였습니다.
- [X] 코드 컨벤션을 모두 확인하였습니다.
- [X] 브랜치명을 확인하였습니다.

Eun0713 committed:
chore: HLS.js 라이브러리 설치

Eun0713 committed:
feat: 스트리밍 재생/일시정지 유틸 함수 추가

Eun0713 committed:
feat: ChannelPlayer에 스트리밍 제어 기능 연결

Eun0713 committed:
feat: MiniPlayer에 스트리밍 제어 기능 연결

devjaeyoon reviewed (dismissed):
LGTM!

Eun0713 committed:
Merge branch 'dev' into feat/#44

faber-h commented on src/utils/playControl.js:
```diff
@@ -0,0 +1,27 @@
+let hlsInstance = null;
+
+const controlStreamPlayback = (videoId, isPlaying) => {
+  const video = document.getElementById(videoId);
```
videoId 값을 잘 가져와주셨네요!
다만, 언어가 React니까 useRef로 사용할 수 있으면 좋을 것 같아요~

d0-neii replied:
네! 말씀 주신 대로 `useRef`가 React에 더 자연스러워서 해당 부분은 `ref`로 수정했습니다😊 좋은 피드백 감사합니다:)

devjaeyoon committed:
feat: 신규 사용자 등록 API 연동 (#49)

* feat: useUserId 훅 구현 및 Home 페이지에 적용

- localStorage에 기존 userId가 있는지 확인
- 없을 경우, API를 호출해 신규 사용자 등록
- 발급받은 userId를 localStorage에 저장
- Home 페이지 진입 시 useUserId 훅 호출

* feat: useUserId 훅에 try-catch를 추가해 에러 처리

- API 호출 부분을 try-catch로 감싸 예외 상황 처리
- 실패 시 콘솔에 에러 메시지 출력

* feat: useUserId 반환값 userId에 저장 및 임시로 useCategorizeChannels에 매개변수로 전달

Eun0713 committed:
refactor: DOM 직접 조작 로직을 ref 기반 처리로 변경

Eun0713 committed:
refactor: MiniPlayer의 DOM 조작 방식을 ref 기반으로 수정

Eun0713 committed:
rename: controlStreamPlayback → controlStreamingPlayback 네이밍 개선

faber-h reviewed (approved):
LGTM🔥
